ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -699,6 +699,8 @@ FindInfo findEntriesSince(FindToken token, long maxTotalSizeOfEntries) throws St
       Offset logEndOffsetBeforeFind = log.getEndOffset();
       StoreFindToken storeToken = resetTokenIfRequired((StoreFindToken) token);
       logger.trace("Time used to validate token: {}", (time.milliseconds() - startTimeInMs));
+      ConcurrentSkipListMap<Offset, IndexSegment> indexSegments = validIndexSegments;
+      storeToken = revalidateStoreFindToken(storeToken, indexSegments);
```

Comment:
`resetTokenIfRequired()` is done only once at the beginning of the function. Revalidating store find token however is done whenever a new view is obtained. So they should be distinct functions.